### PR TITLE
Remove old fd init code path

### PR DESCRIPTION
**Pull request by tt-dma**
New FD init has landed and seems stable, this PR removes the old FD init.

CI: https://github.com/tenstorrent/tt-metal/actions/runs/12489907171, https://github.com/tenstorrent/tt-metal/actions/runs/12495980066 (same failures as main)

### CODE REVIEW

File: tt_metal/impl/device/device.cpp [hunk 1]
```diff
@@ -3980,35 +1867,6 @@ std::vector<CoreCoord> Device::get_optimal_dram_bank_to_logical_worker_assignmen
     return this->optimal_dram_bank_to_logical_worker_assignment_;
 }
 
-HalProgrammableCoreType Device::get_programmable_core_type(CoreCoord virtual_core) const {
```
Comment (omilyutin-tt) [inline tt_metal/impl/device/device.cpp:3983]: This PR undoes (some of? all?) changes @ayerofieiev-tt made in https://github.com/tenstorrent/tt-metal/pull/16259 -- is this intentional, or bad rebase?

**Review by omilyutin-tt — CHANGES_REQUESTED** [review]

File: tt_metal/impl/device/device.cpp [hunk 2]
```diff
@@ -3691,18 +1582,14 @@ void Device::enable_async(bool enable) {
...
-    return using_fast_dispatch_;
+    return not (this->using_fast_dispatch_);
 }
 
 void Device::begin_trace(const uint8_t cq_id, const uint32_t tid) {
     ZoneScoped;
     TracyTTMetalBeginTrace(this->id(), tid);
     TT_FATAL(!this->hw_command_queues_[cq_id]->tid.has_value(), "CQ {} is already being used for tracing tid {}", (uint32_t)cq_id, tid);
-    this->mark_allocations_safe();
+    this->MarkAllocationsSafe();
```
Comment (arikTT) [inline tt_metal/impl/device/device.cpp:1592]: why are we changing casing? Even for camel case, first `m` should be lower case

**Review by arikTT — COMMENTED** [review]
I love PRs that remove code!

File: tt_metal/impl/device/device.cpp (hunk 1 above)
Comment (tt-dma) [reply tt_metal/impl/device/device.cpp:3983]: Bad rebase I think, let me check on this

File: tt_metal/impl/device/device.cpp (hunk 2 above)
Comment (tt-dma) [reply tt_metal/impl/device/device.cpp:1592]: Bad rebase I think, will fix

**Comment by tt-dma** [comment]
Fixed a bad rebase, should be ok to look at now

**Review by omilyutin-tt — APPROVED** [review]
Impressive diff, this is great!

File: tt_metal/impl/device/device.cpp (hunk 1 above)
Comment (tt-dma) [reply tt_metal/impl/device/device.cpp]: Should be fixed now

File: tt_metal/impl/device/device.cpp (hunk 2 above)
Comment (tt-dma) [reply tt_metal/impl/device/device.cpp]: Should be fixed now

**Review by ayerofieiev-tt — APPROVED** [review]
big win!

**Review by arikTT — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]

**Review by tt-asaigal — APPROVED** [review]

**Review by abhullar-tt — APPROVED** [review]

**Comment by tt-dma** [comment]
New passing CI: https://github.com/tenstorrent/tt-metal/actions/runs/12519558951, https://github.com/tenstorrent/tt-metal/actions/runs/12519561238 (same failure as main)

**Pull request merged**